synthtool/sources/git.py

```diff
@@ -82,9 +86,9 @@ def clone(
             cmd = ["git", "clone", "--recurse-submodules", "--single-branch", url, dest]
             shell.run(cmd, check=True)
         else:
-            shell.run(["git", "checkout", "master"], cwd=str(dest), check=True)
+            shell.run(["git", "checkout", default_branch], cwd=str(dest), check=True)
```

Comment:
I think we could do language maintainers a favor by detecting the default branch here.  Otherwise they'll have to update code after they rename the branch.

I believe github renames the branch, so that after set the default branch to `main`, there is no more `master` branch.  Also, we clone the repo with `--single-branch` on line 86 so there should only be one branch.  Therefore, we could list the branches in cwd and see which branch name is present:
```
git branch --sort=committerdate
```

The `--sort=commiterdate` will break the tie if both names appear.  We'll take the one with the more recent commits.

Reply:
Added a helper function to detect the latest branch